--daemon --log-level=panic panic

Hello,

Using  `--daemon --log-level=panic` with the last RC produce a panic.

```
$ ./docker-1.7.0-rc1  --daemon --log-level=panic
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x40c1de]

goroutine 1 [running]:
main.func·002()
	/go/src/github.com/docker/docker/docker/daemon.go:103 +0xfe
main.mainDaemon()
	/go/src/github.com/docker/docker/docker/daemon.go:158 +0x1352
main.main()
	/go/src/github.com/docker/docker/docker/docker.go:92 +0x6b4

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 8 [runnable]:
text/template/parse.lexText(0xc208030500, 0xf38af8)
	/usr/local/go/src/text/template/parse/lex.go:228 +0x37b
text/template/parse.(*lexer).run(0xc208030500)
	/usr/local/go/src/text/template/parse/lex.go:198 +0x5d
created by text/template/parse.lex
	/usr/local/go/src/text/template/parse/lex.go:191 +0x1ac

goroutine 9 [runnable]:
main.func·003()
	/go/src/github.com/docker/docker/docker/daemon.go:128
created by main.mainDaemon
	/go/src/github.com/docker/docker/docker/daemon.go:135 +0x6bc
```

--- a/hack/vendor.sh
+++ b/hack/vendor.sh
@@ -43,7 +43,7 @@ clone() {
 }
 
 # the following lines are in sorted order, FYI
-clone git github.com/Sirupsen/logrus v0.7.3 # logrus is a common dependency among multiple deps
+clone git github.com/Sirupsen/logrus v0.8.2 # logrus is a common dependency among multiple deps
 clone git github.com/docker/libtrust 230dfd18c232
 clone git github.com/go-check/check 64131543e7896d5bcc6bd5a76287eb75ea96c673
 clone git github.com/gorilla/context 14f550f51a

